src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +465,152 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory
+     * @param newFileName the new directory name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void renameDirectory(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Yes, maybe you are correct. I tried to use `setRenameFileInfomation(String path, String newFileName, boolean replaceIfExist, EnumSet<SMB2CreateOptions> createOptions)` wtih createOptions null and it can still rename the file and the directory. I think we can simply having the `rename (Directory | File)` for the rename case.

The reason why I try to distinguish them is because I worked in another project written in C for Windows files handling and I remember some bugs are cause by mix-up the directory and file caused so much trouble. So, I just want to place safe and set the `FILE_DIRECTORY_FILE` flag for directory and `FILE_NON_DIRECTORY_FILE` flag for file when opening them (So that the server will return NTSTATUS `STATUS_NOT_A_DIRECTORY` or `STATUS_FILE_IS_A_DIRECTORY`). But seems this not applied to SMB2 rename case by my testing result. So, I think we can ignore the differences for the rename case.

I would combine the functions as `rename(Directory | File)`.